src/app/Main/Body/QueryPage/dataApi.service.ts

```diff
@@ -0,0 +1,41 @@
+import { adaptIsolatesFromAPI } from "../../../Core/adaptIsolatesFromAPI.service";
+import { FilterConfigDTO } from "../../../Shared/Model/Api_Filter.model";
+import { IsolateDTO } from "../../../Shared/Model/Api_Isolate.model";
+import { DbCollection } from "../../../Shared/Model/Client_Isolate.model";
+import { FilterInterface } from "../../../Shared/Model/Filter.model";
+import { generateUniqueValuesService } from "./generateUniqueValues.service";
+
+export async function dataApiService(

```

Comment:
Recommendations
- don't merge the different api endpoints
- i would use the urls right here, no need to use them as a parameter
- use a generic type for the response:
```
interface ApiResponse<T extends {}>{
  status: number,
  data?: T
}
```
- as for now the api services do not really do something useful, you've could have placed the stuff to your mapping services and just renamed them: adaptIsolatesFromApi => fetchIsolatesFromApi. If you extract the fetching and jsoning to a general api service this would become more obvious.

Reply:
Improved in commit "Change ApiServices"